Firestore/Source/Model/FSTFieldValue.mm

```diff
@@ -225,26 +238,24 @@ - (FSTTypeOrder)typeOrder {
 }
 
 - (NSComparisonResult)compare:(FSTFieldValue *)other {
-  if (![other isKindOfClass:[FSTNumberValue class]]) {
+  if (!([other type] == FieldValue::Type::Integer || [other type] == FieldValue::Type::Double)) {
```

Comment:
Maybe extract this into a `static bool FieldValue::IsNumber(Type)`?

Reply:
Done.